local-cli/runIOS/runIOS.js

```diff
@@ -232,9 +244,11 @@ function printFoundDevices(devices) {
   }
 }
 
-function getProcessOptions(launchPackager) {
+function getProcessOptions(launchPackager, port=null) {
   if (launchPackager) {
-    return {};
+    return {
+      env: Object.assign({}, process.env, { RCT_PACKAGER: port } )
```

Comment:
Nit: We can use object spread instead of Object.assign here:
`return {env: {...process.env, RCT_PACKAGER: port}};`